tests/tests/signer_jwt_auth.rs

```diff
@@ -104,3 +106,115 @@ async fn test_signer_jwt_rate_limit() -> Result<()> {
+
+    // After revoke, test module shouldn't be allowed anymore
+    let response = client.get(&url).bearer_auth(&jwt).send().await?;
+    assert!(response.status() == StatusCode::UNAUTHORIZED);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_signer_only_admin_can_revoke() -> Result<()> {
+    setup_test_env();
+    let admin_secret = ADMIN_SECRET.to_string();
+    let module_id = ModuleId(JWT_MODULE.to_string());
+    let mod_cfgs = create_mod_signing_configs().await;
+    let start_config = start_server(20500, &mod_cfgs, admin_secret.clone()).await?;
+
+    // Run as many pubkeys requests as the fail limit
+    let jwt = create_jwt(&module_id, JWT_SECRET)?;
+    let admin_jwt = create_admin_jwt(admin_secret)?;
+    let client = reqwest::Client::new();
+    let url = format!("http://{}{}", start_config.endpoint, REVOKE_MODULE_PATH);
+
+    // Module JWT shouldn't be able to revoke modules
+    let response = client
+        .post(&url)
+        .header("content-type", "application/json")
+        .body(reqwest::Body::wrap(format!("{{\"module_id\": \"{JWT_MODULE}\"}}")))
+        .bearer_auth(&jwt)
+        .send()
+        .await?;
+    assert!(response.status() == StatusCode::UNAUTHORIZED);
+
+    // Admin should be able to revoke modules
+    let response = client
+        .post(&url)
+        .header("content-type", "application/json")
+        .body(reqwest::Body::wrap(format!("{{\"module_id\": \"{JWT_MODULE}\"}}")))
+        .bearer_auth(&admin_jwt)
+        .send()
+        .await?;
+    assert!(response.status() == StatusCode::OK);
+
+    Ok(())
+}
+
+// Starts the signer moduler server on a separate task and returns its
+// configuration
+/*
+async fn start_server(port: u16) -> Result<StartSignerConfig> {
+    setup_test_env();
+    let chain = Chain::Hoodi;
+
+    // Mock JWT secrets
+    let module_id = ModuleId(JWT_MODULE.to_string());
+    let mut jwts = HashMap::new();
+    jwts.insert(module_id.clone(), JWT_SECRET.to_string());
+
+    // Create a signer config
+    let loader = SignerLoader::ValidatorsDir {
+        keys_path: "data/keystores/keys".into(),
+        secrets_path: "data/keystores/secrets".into(),
+        format: ValidatorKeysFormat::Lighthouse,
+    };
+    let mut config = get_signer_config(loader);
+    config.port = port;
+    config.jwt_auth_fail_limit = 3; // Set a low fail limit for testing
+    config.jwt_auth_fail_timeout_seconds = 3; // Set a short timeout for testing
+    let start_config = get_start_signer_config(config, chain, jwts, ADMIN_SECRET.to_string());
+
+    // Run the Signer
+    let server_handle = tokio::spawn(SigningService::run(start_config.clone()));
+
+    // Make sure the server is running
+    tokio::time::sleep(Duration::from_millis(100)).await;
+    if server_handle.is_finished() {
+        return Err(eyre::eyre!(
+            "Signer service failed to start: {}",
+            server_handle.await.unwrap_err()
+        ));
+    }
+    Ok(start_config)
+}
+
+*/
```

Comment:
Is this intended?

Reply:
Nope, was just leftover from the merge - the server test code with `signing_id` support is now here: https://github.com/Commit-Boost/commit-boost-client/blob/688d82d11e07acca11925f575b1c68e31af5cb0a/tests/src/signer_service.rs#L20

I removed this comment block in the latest, thanks for catching it!